lib/lte_link_control/lte_lc.c

```diff
@@ -1536,33 +1388,60 @@ int lte_lc_conn_eval_params_get(struct lte_lc_conn_eval_params *params)
 		return -EOPNOTSUPP;
 	}
 
-	/* Read connection evaluation parameters. */
-	err = at_cmd_write(AT_CONEVAL_READ, buf, sizeof(buf), NULL);
+	/* AT%CONEVAL response format, from nRF91 AT Commands - Command Reference Guide, v1.7:
+	 *
+	 * %CONEVAL: <result>[,<rrc_state>,<energy_estimate>,<rsrp>,<rsrq>,<snr>,<cell_id>,<plmn>,
+	 * <phyc_cell_id>,<earfcn>,<band>,<tau_triggered>,<ce_level>,<tx_power>,<tx_repetitions>,
+	 * <rx_repetitions>,<dl-pathloss>]
+	 *
+	 * In total, 17 parameters are expected to match for a successful command response.
+	 */
+	err = nrf_modem_at_scanf(
+		AT_CONEVAL_READ,
+		"%%CONEVAL: %u,%u,%u,%d,%d,%u,\"%x\",\"%6s\",%u,%u,%u,%u,%u,%d,%u,%u,%d",
+		&result, &params->rrc_state, &params->energy_estimate,
+		&params->rsrp, &params->rsrq, &params->snr, &params->cell_id,
+		plmn_str, &params->phy_cid, &params->earfcn, &params->band,
+		&params->tau_trig, &params->ce_level, &params->tx_power,
+		&params->tx_rep, &params->rx_rep, &params->dl_pathloss);
```

Comment:
Ah yes, that's because I set the PLMN to always be 6 characters. Changed this now to be maximum 6, and verified that it works with the response that you received.
Looking into adding tests with mocked `nrf_modem_at_` calls to get test coverage in this.